src/fastgltf.cpp

```diff
@@ -1303,366 +1313,445 @@ fg::Expected<fg::Asset> fg::Parser::parse(simdjson::dom::object root, Category c
         }
 
-        std::uint64_t componentType;
-        if (accessorObject["componentType"].get_uint64().get(componentType) != SUCCESS) {
-            return Error::InvalidGltf;
-        }
-		accessor.componentType = getComponentType(static_cast<std::underlying_type_t<ComponentType>>(componentType));
-        if (accessor.componentType == ComponentType::Double && !hasBit(options, Options::AllowDouble)) {
-            return Error::InvalidGltf;
-        }
-
-        std::string_view accessorType;
-        if (accessorObject["type"].get_string().get(accessorType) != SUCCESS) {
-            return Error::InvalidGltf;
-        }
-		accessor.type = getAccessorType(accessorType);
-
-        std::uint64_t accessorCount;
-        if (accessorObject["count"].get_uint64().get(accessorCount) != SUCCESS) {
-            return Error::InvalidGltf;
-        }
-		accessor.count = static_cast<std::size_t>(accessorCount);
-
-
-        std::uint64_t bufferView;
-        if (accessorObject["bufferView"].get_uint64().get(bufferView) == SUCCESS) {
-            accessor.bufferViewIndex = static_cast<std::size_t>(bufferView);
-        }
+        auto& accessor = asset.accessors.emplace_back();
+        accessor.byteOffset = 0UL;
+        accessor.normalized = false;
 
-        // byteOffset is optional, but defaults to 0
-        std::uint64_t byteOffset;
-        if (auto error = accessorObject["byteOffset"].get_uint64().get(byteOffset); error == SUCCESS) {
-            accessor.byteOffset = static_cast<std::size_t>(byteOffset);
-        } else if (error != NO_SUCH_FIELD) {
-            return Error::InvalidGltf;
-        }
+        // Function for parsing the min and max arrays for accessors
+        auto parseMinMax = [&](simdjson::ondemand::array& array, decltype(Accessor::max)& ref) -> fastgltf::Error {
+			return Error::None;
+            decltype(Accessor::max) variant;
 
-        // Type of min and max should always be the same.
-        auto parseMinMax = [&](std::string_view key, decltype(Accessor::max)& ref) -> fastgltf::Error {
-            dom::array elements;
-            if (accessorObject[key].get_array().get(elements) == SUCCESS) {
-                decltype(Accessor::max) variant;
+            using double_vec = std::variant_alternative_t<1, decltype(Accessor::max)>;
+            using int64_vec = std::variant_alternative_t<2, decltype(Accessor::max)>;
 
-				using double_vec = std::variant_alternative_t<1, decltype(Accessor::max)>;
-				using int64_vec = std::variant_alternative_t<2, decltype(Accessor::max)>;
+            // It's possible the min/max fields come before the accessor type is declared, in which case we don't know the size.
+			// This single line is actually incredibly important as this function without it takes up roughly 15% of the entire
+			// parsing process on average due to vector resizing.
+            auto initialCount = accessor.type == AccessorType::Invalid ? 0 : getNumComponents(accessor.type);
```

Comment:
Is initialCount used anywhere?

Reply:
yeah, oops. initialCount should be used two lines further down for reserving the vector capacity:
```cpp
auto vec = FASTGLTF_CONSTRUCT_PMR_RESOURCE(double_vec, resourceAllocator.get(), 0);
vec.reserve(initialCount);
variant = std::move(vec);
```